### ISSUE TITLE
CVE-2023-42503 - Improper Input Validation vulnerability in org.apache.commons:commons-compress

### ISSUE THREAD

**Issue by abstractj** [keycloak/keycloak#23331]
# Overview

[org.apache.commons:commons-compress](https://github.com/apache/commons-compress) is an API for working with compression and archive formats.

Affected versions of this package are vulnerable to Improper Input Validation when parsing TAR files. An attacker can manipulate file modification times headers, leading to a denial of service issue via CPU consumption.

**Note:** This is only exploitable if applications are using the `CompressorStreamFactory` class (with auto-detection of file types), `TarArchiveInputStream`, and `TarFile` classes to parse TAR files.

# Remediation

Upgrade `org.apache.commons:commons-compress` to version 1.24.0 or higher.

# References

- [Apache Advisory](https://lists.apache.org/thread/5xwcyr600mn074vgxq92tjssrchmc93c)
- [GitHub Commit](https://github.com/apache/commons-compress/commit/43354810a9aa4305bf82318f1884e83a078a621a)

# Additional comments

- After analyzing the issue, it appears that the transitive dependency comes from the following dependency tree::

```
+- io.fabric8:kubernetes-junit-jupiter:jar:6.7.2:test
[INFO] |  +- io.fabric8:kubernetes-client:jar:6.7.2:compile
[INFO] |  |  +- io.fabric8:kubernetes-httpclient-okhttp:jar:6.7.2:runtime
[INFO] |  |  |  +- com.squareup.okhttp3:okhttp:jar:3.14.9:runtime
[INFO] |  |  |  |  \- com.squareup.okio:okio:jar:1.17.2:runtime
[INFO] |  |  |  \- com.squareup.okhttp3:logging-interceptor:jar:3.14.9:runtime
[INFO] |  |  \- io.fabric8:zjsonpatch:jar:0.3.0:compile
[INFO] |  \- org.apache.commons:commons-compress:jar:1.23.0:compile
```

As we discussed earlier, `kubernetes-junit-jupiter` is only for testing purposes. However, the package is included in the current image of the Operator. For more details, refer to the Trivy vulnerability report below:

```
❯ trivy image -f json --timeout 25m --quiet --ignore-unfixed quay.io/keycloak/keycloak-operator:22.0.3 | jq '.Results[].Vulnerabilities[]? | "\(.VulnerabilityID) - \(.PkgName) - \(.PkgPath) - \(.Title)"'
"CVE-2023-42503 - org.apache.commons:commons-compress - opt/keycloak/lib/main/org.apache.commons.commons-compress-1.23.0.jar - Improper Input Validation, Uncontrolled Resource Consumption vulnerabi ..."
```

**Comment by vmuzikar** [comment]
@shawkins Can you please look into it when you get a chance? Possibly it's something that should be fixed on Quarkus side once they upgraded the deps in their BOM.

**Comment by shawkins** [comment]
@vmuzikar if this was still appearing in our image, it should be addressed by #22496 as commons-compress is an optional dependency.  The fabric8 client only uses that for exec file transfer operations.  When I locally build main or release/22.0 commons-compress the dependency is not included in the lib/main - it is however still listed in the quarkus-app-dependencies.txt, which may be throwing off the analysis.

**Comment by shawkins** [comment]
Marking as resolved based upon the previous comment.  There may need to be a separate follow-up with the analysis tool so that it better understands test and/or filtered optional dependencies.

**Comment by shawkins** [comment]
@pgodowski I must have been looking for the jar in the lib based upon the naming convention for commons-codec.  I see the issue now.   The kubernetes-junit-jupiter (a test scope dependency) is marking commons-compress as non-optional and that is somehow overriding the optional flag for commons-compress at a runtime scope.  That feels like a maven bug, but we can easily work around it.

**Comment by pgodowski** [comment]
Thanks a lot for addressing this.

Sorry for perhaps a stupid question, @vmuzikar .

I see the issue fixed in `quay.io/keycloak/keycloak-operator:nightly`, but it's still there in `quay.io/keycloak/keycloak-operator:22.0`.

Are the nightly builds made off the main branch (i.e. v23) or is there also nightly build of 22.0.next available, based on `release/22.0` branch too (don't see such images it on `quay.io/repository/keycloak/keycloak-operator` )?

**Comment by vmuzikar** [comment]
There's no nightlies from the release branches (e.g. `release/22.0`) at the moment, I'm afraid.
